File: OCH_Tao/ABC/B/294.py
from string import ascii_uppercase
S = "."+ascii_uppercase
ans = ""
H, W = map(int, input().split())
for i in range(H):
  A = list(map(int, input().split()))
  ans += "".join(map(lambda x: S[x], A))+"\n"
print(ans, end="")

File: OCH_Tao/ABC/B/294.js
const [, ...A] = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n").map(x => x.split(" ").map(y => parseInt(y, 10)));
let ans = [];
for (const i of A) {
  ans.push(i.map(x => x === 0 ? "." : String.fromCharCode(64 + x)).join(""));
}
console.log(ans.join("\n"));